src/lxc/conf.c

```diff
@@ -3382,21 +3449,21 @@ int lxc_map_ids(struct lxc_list *idmap, pid_t pid)
 			continue;
 
 		if (!use_shadow) {
-			ret = write_id_mapping(type, pid, buf, pos - buf);
+			fret = write_id_mapping(type, pid, buf, pos - buf);
 		} else {
 			left = LXC_IDMAPLEN - (pos - buf);
 			fill = snprintf(pos, left, "\n");
 			if (fill <= 0 || fill >= left)
 				SYSERROR("Too many {g,u}id mappings defined.");
 			pos += fill;
-			ret = system(buf);
+			fret = system(buf);
 		}
 		if (ret)
```

Comment:
`ret` is used in the `new{g,u}idmap` block before and might get initialized to a non-zero value but the `for`-loop, as unlikely as this is, might not be run and then you'd exit with error.